libs/wingc/src/diagnostic.rs

```diff
@@ -172,6 +172,58 @@ impl PartialOrd for Diagnostic {
 	}
 }
 
+thread_local! {
+	pub static DIAGNOSTICS: RefCell<Diagnostics> = RefCell::new(Diagnostics::new());
+}
+
+/// Report a compilation diagnostic
+pub fn report_diagnostic(diagnostic: Diagnostic) {
+	// Add the diagnostic to the list of diagnostics
+	DIAGNOSTICS.with(|diagnostics| {
+		diagnostics.borrow_mut().push(diagnostic.clone());
+	});
+
+	// If we're running in wasm32 then send the diagnostic to the client
+	#[cfg(target_arch = "wasm32")]
+	{
+		let json = serde_json::to_string(&diagnostic).unwrap();
+		let bytes = json.as_bytes();
+		unsafe {
+			new_diagnostic(bytes.as_ptr(), bytes.len() as u32);
+		}
+	}
+}
+
+#[cfg(target_arch = "wasm32")]
+extern "C" {
+	pub fn new_diagnostic(data: *const u8, data_length: u32);
+}
+
+/// Returns whether any errors were found during compilation
+pub fn found_errors() -> bool {
+	DIAGNOSTICS.with(|diagnostics| {
+		let diagnostics = diagnostics.borrow();
+		!diagnostics.is_empty()
+	})
+}
+
+/// Returns the list of diagnostics
+pub fn get_diagnostics() -> Vec<Diagnostic> {
```

Comment:
nit: Possible to avoid cloning for our use cases?

Reply:
not sure how, I don't think I can "move out" of the thread local storage and the users here need a clone anyway (the diags are sorted or copied into FileData ...).